### PR TITLE
Hook Linting

### PR DESCRIPTION
**Pull request by rudle**
Do not review yet.

### CODE REVIEW

**Comment by rudle** [comment]
OK. This is ready for review. No functionality should be changed by this.

The commits are not super clean, I'm sorry about that.

File: pkg/hooks/hooks_test.go [hunk 1]
```diff
@@ -30,10 +30,10 @@ func TestExecutableHooksAreRun(t *testing.T) {
 	// So PodFromPodHome doesn't bail out, write a minimal current_manifest.yaml
 	ioutil.WriteFile(path.Join(podDir, "current_manifest.yaml"), []byte("id: my_hook"), 0755)
 
-	hooks := Hooks(os.TempDir(), pods.DefaultPath, &logging.DefaultLogger)
+	hooks := NewContext(tempDir, pods.DefaultPath, &logging.DefaultLogger)
```
Comment (rudle) [inline pkg/hooks/hooks_test.go]: This is a curious change!

File: pkg/hooks/types.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+)
+
+// HookType represents the stage in the Hook lifecycle
+type HookType string
+
+func (hookType HookType) String() string {
+	return string(hookType)
+}
+
+var (
```
Comment (petertseng) [inline pkg/hooks/types.go]: so, while yo'ure moving these - any chance of making these `const`s? Doesn't seem we'd want to change them.

File: pkg/hooks/types.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+		return AfterLaunch, nil
+	case AfterAuthFail.String():
+		return AfterAuthFail, nil
+	default:
+		return HookType(""), fmt.Errorf("%s is not a valid hook type", value)
+	}
+}
+
+// DefaultPath is a directory on disk where hooks are installed by default.
+var DefaultPath = "/usr/local/p2hooks.d"
```
Comment (petertseng) [inline pkg/hooks/types.go]: should this be const? Or do we want to keep as var to allow override at build-time?

File: pkg/hooks/types.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+	podRoot string
+	logger  *logging.Logger
+}
+
+type hookExecContext struct {
+	Path    string // path to hook's executable
+	Name    string // human-readable name of Hook
+	Timeout time.Duration
+	env     []string // unix environment in which the hook should be executed
+	logger  *logging.Logger
```
Comment (petertseng) [inline pkg/hooks/types.go]: change to pointer noted

**Pull request merged**